src/tools/r2rdump/R2RReader.cs

```diff
@@ -81,63 +81,69 @@ public unsafe R2RReader(string filename)
                 R2RHeader = new R2RHeader(_image, r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
                 if (r2rHeaderDirectory.Size != R2RHeader.Size)
                 {
-                    throw new System.BadImageFormatException("The calculated size of the R2RHeader doesn't match the size saved in the ManagedNativeHeaderDirectory");
+                    throw new BadImageFormatException("The calculated size of the R2RHeader doesn't match the size saved in the ManagedNativeHeaderDirectory");
                 }
 
                 if (peReader.HasMetadata)
                 {
                     var mdReader = peReader.GetMetadataReader();
 
+                    int runtimeFunctionSize = 2;
+                    if (Machine == Machine.Amd64)
+                    {
+                        runtimeFunctionSize = 3;
+                    }
+                    runtimeFunctionSize *= sizeof(int);
+                    R2RSection runtimeFunctionSection = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_RUNTIME_FUNCTIONS];
+                    uint nRuntimeFunctions = (uint)(runtimeFunctionSection.Size / runtimeFunctionSize);
+                    int runtimeFunctionOffset = GetOffset(runtimeFunctionSection.RelativeVirtualAddress);
+                    bool[] isEntryPoint = new bool[nRuntimeFunctions];
+                    for (int j = 0; j < nRuntimeFunctions; j++)
+                    {
+                        isEntryPoint[j] = false;
+                    }
```

Comment:
I've made some changes to how the methods get matched to runtime functions to make sure it uses the runtime function id as entrypoint, and gets all the functions until the next entrypoint function which might not be the next element of the R2RMethods array